projects/vdk-plugins/vdk-csv/tests/functional/test_csv_plugin.py

```diff
@@ -76,3 +81,114 @@ def test_ingestion_csv_with_options():
 
     assert len(ingest_plugin.payloads) > 0
     assert ingest_plugin.payloads[0].payload[0]["US Zip"] == "08056"
+
+
+def test_csv_export(tmpdir):
+    db_dir = str(tmpdir) + "vdk-sqlite.db"
+    with mock.patch.dict(
+        os.environ,
+        {
+            "VDK_DB_DEFAULT_TYPE": "SQLITE",
+            "VDK_SQLITE_FILE": db_dir,
+        },
+    ):
+        runner = CliEntryBasedTestRunner(sqlite_plugin, csv_plugin)
+        runner.invoke(
+            [
+                "sqlite-query",
+                "--query",
+                "CREATE TABLE test_table (some_data TEXT, more_data TEXT)",
+            ]
+        )
+
+        mock_sqlite_conf = mock.MagicMock(SQLiteConfiguration)
+        sqlite_ingest = IngestToSQLite(mock_sqlite_conf)
+        payload = [
+            {"some_data": "some_test_data", "more_data": "more_test_data"},
+            {"some_data": "some_test_data_copy", "more_data": "more_test_data_copy"},
+        ]
+
+        sqlite_ingest.ingest_payload(
+            payload=payload,
+            destination_table="test_table",
+            target=db_dir,
+        )
+
+        runner.invoke(["export-csv", "--query", "SELECT * FROM test_table"])
+        output = []
+        with open(os.path.join(os.path.abspath(os.getcwd()), "result.csv")) as file:
+            reader = csv.reader(file, delimiter=",")
+            for row in reader:
+                output.append(row)
+        assert output[0] == ["some_test_data", "more_test_data"]
+        assert output[1] == ["some_test_data_copy", "more_test_data_copy"]
+
+
+def test_export_csv_with_already_existing_file(tmpdir):
+    db_dir = str(tmpdir) + "vdk-sqlite.db"
+    with mock.patch.dict(
+        os.environ,
+        {
+            "VDK_DB_DEFAULT_TYPE": "SQLITE",
+            "VDK_SQLITE_FILE": db_dir,
+        },
+    ):
+        path = os.path.abspath(os.getcwd())
+        open(os.path.join(path, "result2.csv"), "w")
+        runner = CliEntryBasedTestRunner(csv_plugin)
+        result = runner.invoke(
+            [
+                "export-csv",
+                "--query",
+                "SELECT * FROM test_table",
+                "--name",
+                "result2.csv",
+            ]
+        )
+        assert isinstance(result.exception, UserCodeError)
+        cli_assert_equal(1, result)
+
+
+def test_csv_export_with_no_data(tmpdir):
```

Comment:
I don't think export-csv should fail if there's no data. 

A) If the SQL query returns empty result (or teh table is empty) , no rows -> we should return empty file. It's normal for people to write query that may return empty result. If this is an issue or not is not something we can tell. 


B) if the table doesn't exist, then we should fail  

Let's split this test into 2 to handle both cases and fix code accordingly.